### DISCUSSION TITLE
Support partial directory by tags

### DISCUSSION THREAD

**Discussion by almousa1990** [harttle/liquidjs#309] [Ideas]
Hello,

I would like to thank the maintainer for the great efforts been made with this repository. I have been trying to find a mechanism resolve template files based on the tag in which the file is read. For example, it doesn't make sense for layout tag to resolve templates under partials directory.

It looks like that file parser will try to resolve all roots defined within liquid opts, regardless of template type, from my understanding, this behaviour can be altered by overriding the tag parser and changing opts root to the desired directory.

**Comment by harttle** [comment]
> resolve templates under partials directory.

I regard it as a badly named option. It's a lot easier if that's your only concern. Could you provide a specific use case? I'm not sure whether you want to load files dynamically based on lots of custom tags, or just want a separate path for `layout` (and keep `partials` for `render`/`include`).

**Comment by almousa1990** [comment]
Thank you for the prompt response, a separate path for templates, layout, partials would be necessary especially when you have different templates with the same name under different directories.

Also, when looking at shopify implementation, they have a separate section directory for section tags.

Maybe we can just have a mechanism to override tag render by providing custom root, and liquid optional root for builtin layout, include/render

**Reply by harttle** [reply]
Maybe it's clearer to provide a `root` option for each of the `layout`, `render` and `include` tags. Let's see if others also need this feature.
